armi/physics/neutronics/fissionProductModel/lumpedFissionProduct.py

```diff
@@ -69,46 +67,45 @@ def __init__(self, name=None):
         """
         self.name = name
         self.yld = {}
-        self.gasRemainingFrac = 1.0
 
     def duplicate(self):
         """
         Make a copy of this w/o using deepcopy
         """
         new = self.__class__(self.name)
-        new.gasRemainingFrac = self.gasRemainingFrac
         for key, val in self.yld.items():
             new.yld[key] = val
         return new
 
-    def __getitem__(self, fissionProduct, default=None):
-        r"""
-        Return the FP yield of a particular FP
+    def __getitem__(self, fissionProduct):
+        """
+        Return the yield of a particular fission product.
 
         This allows the LFP to be accessed via indexing, like this: ``lfp[fp]``
 
         Returns
         -------
-        yld : yield of the fission product. Defaults to None.
+        yld : yield of the fission product.
         """
-        yld = self.yld.get(fissionProduct, default)
-        if yld and isGas(fissionProduct):
-            yld *= self.gasRemainingFrac
-        return yld
```

Comment:
This was modified and replaced with the number density updates implemented in `removeFissionGasesFromBlocks` on the `FissionProductModel` interface. I cannot speak to why the yields were being updated but I believe this to be incorrect.

Essentially, the old implementation modified the yield fractions of gaseous nuclides in the lumped fission products rather than modifying the number densities directly in the blocks/components. In this way, the global flux calculation would have number densities that would not be reflected by the removal of the gaseous fission products. Instead, when `MC2-3` code was run, the microscopic XS for `LFP35`, `LFP38`, etc. were being augmented. 

By modifying the number densities directly this will capture both the cross section AND the flux solution effects together rather than only sort of treating one of the effects.